internal/controller/datadogagent/controller_reconcile_v2_common.go

```diff
@@ -629,6 +632,77 @@ func (r *Reconciler) addDDAIStatusToDDAStatus(status *datadoghqv2alpha1.DatadogA
 	return nil
 }
 
+// getCurrentDeployment returns the current deployment for a given DDA
+func (r *Reconciler) getCurrentDeployment(dda, deployment metav1.Object) (*appsv1.Deployment, error) {
+	// Helm-migrated deployment
+	if helm.IsHelmMigration(dda) {
+		componentType := deployment.GetLabels()[apicommon.AgentDeploymentComponentLabelKey]
+		if componentType == "" {
+			r.log.Info("No component label found in deployment, using default")
+			componentType = constants.DefaultAgentResourceSuffix
+		}
+		dsList := appsv1.DeploymentList{}
+		matchLabels := client.MatchingLabels{}
+		// Special handling for cluster checks runner - helm has different component name
+		if componentType == constants.DefaultClusterChecksRunnerResourceSuffix {
+			matchLabels = client.MatchingLabels{
+				kubernetes.AppKubernetesManageByLabelKey:  "Helm",
+				kubernetes.AppKubernetesInstanceLabelKey:  dda.GetName(),
+				kubernetes.AppKubernetesComponentLabelKey: "clusterchecks-agent",
+			}
+		} else {
+			matchLabels = client.MatchingLabels{
+				kubernetes.AppKubernetesManageByLabelKey:   "Helm",
+				apicommon.AgentDeploymentNameLabelKey:      dda.GetName(),
+				apicommon.AgentDeploymentComponentLabelKey: componentType,
+			}
+		}
```

Comment:
```suggestion
			matchLabels = client.MatchingLabels{
				kubernetes.AppKubernetesManageByLabelKey:   "Helm",
				apicommon.AgentDeploymentNameLabelKey:      dda.GetName(),
				apicommon.AgentDeploymentComponentLabelKey: componentType,
			}
```

I don't think we need this anymore now that the expected labels have been added to the helm chart